src/cascadia/TerminalApp/CascadiaSettings.cpp

```diff
@@ -452,3 +452,153 @@ void CascadiaSettings::_ValidateAllSchemesExist()
         _warnings.push_back(::TerminalApp::SettingsLoadWarnings::UnknownColorScheme);
     }
 }
+
+// Method Description:
+// - Create a TerminalSettings object for the provided combination of
+//   profileIndex and newTerminalArgs. We'll use the index and newTerminalArgs
+//   to look up the profile that should be used to create these
+//   TerminalSettings. Then, we'll apply settings contained in the
+//   newTerminalArgs to the profile's settings, to enable customization on top
+//   of the profile's default values.
+// Arguments:
+// - index: if provided, the index in the list of profiles to get the GUID for.
+//   If omitted, instead use the default profile's GUID
+// - newTerminalArgs: An object that may contain a profile name or GUID to
+//   actually use. If the Profile value is not a guid, we'll treat it as a name,
+//   and attempt to look the profile up by name instead.
+//   * Additionally, we'll use other values (such as Commandline,
+//     StartingDirectory) in this object to override the settings directly from
+//     the profile.
+// Return Value:
+// - the GUID of the created profile, and a fully initialized TerminalSettings object
+std::tuple<GUID, TerminalSettings> CascadiaSettings::BuildSettings(const NewTerminalArgs& newTerminalArgs) const
+{
+    const GUID profileGuid = _GetProfileForArgs(newTerminalArgs);
+    auto settings = MakeSettings(profileGuid);
+
+    if (newTerminalArgs)
+    {
+        // Override commandline, starting directory if they exist in newTerminalArgs
+        if (!newTerminalArgs.Commandline().empty())
+        {
+            settings.Commandline(newTerminalArgs.Commandline());
+        }
+        if (!newTerminalArgs.StartingDirectory().empty())
+        {
+            settings.StartingDirectory(newTerminalArgs.StartingDirectory());
+        }
+        if (!newTerminalArgs.TabTitle().empty())
+        {
+            settings.StartingTitle(newTerminalArgs.TabTitle());
+        }

```

Comment:
`tabTitle` is the only one it will matter for (and it doesn't work right right now), but this will be lost on a settings reload.

Reply:
That wouldn't matter though, would it? The `startingTitle` doesn't work on reload anyways - it's just the title the process starts with.

Reply:
Er that's what I mean when i say "doesn't work right now" -- `tabTitle` is a weird one where it should totally _win_ if `suppressApplicationTitle` is turned on, right? So we have a minor issue right now where the tab title never gets pushed down to the right place (Terminal Core right now) on _reload_.. but this compounds the issue b/c if we DO implement that properly profiles started with this shortcut will  have their tab titles lost.

Reply:
I added this to _my big ol' list of tasks_ for us to follow up with. We really need to shake this story out, and I think we just need someone to spend a day or two unifying the story here. We have all the right settings at this point, it's just a matter of making sure it's all plumbed through at the right levels.